redash/settings/__init__.py

```diff
@@ -227,6 +227,9 @@ def email_server_is_configured():
 
 HOST = os.environ.get('REDASH_HOST', '')
 
+SEND_FAILURE_EMAIL_INTERVAL = int(os.environ.get('REDASH_SEND_FAILURE_EMAIL_INTERVAL', 3600))
+MAX_FAILURE_REPORTS_PER_QUERY = int(os.environ.get('REDASH_MAX_FAILURE_REPORTS_PER_QUERY', 100))
```

Comment:
Was there a reason that we set it so high? I would expect 10 or so to be enough, no?

Reply:
> Does it make more sense to cap the number of emails sent per query instead of the number of failures per query?

Yep. But how will we track # of emails?

Reply:
Maybe the frequency should be once per day? We can rate limit it by creating an expiring key in Redis for a query id that received a report. Once the key exists -- we stop reporting, when the query expires we resume reporting.

Reply:
Yeah that's tricky because we need some sort of long-term storage (one that doesn't evaporate after an hour) like `Query.scheduled_failures`. We _could_ add an extra column or dump it in `Query.options`, but I'm not sure if we want to invest in that. WDYT?

Reply:
> Yeah that's tricky because we need some sort of long-term storage (one that doesn't evaporate after an hour)

What's wrong with using Redis for this?

Reply:
Keys will be left dangling. We could handle this by setting a weekly/monthly expiry - that is, up to 10 e-mails per query (each e-mail could represent multiple failures), and then silence for the rest of the week/month. Otherwise, the collection of dangling keys will grow and grow.

Reply:
> Keys will be left dangling.

Why? When the query fails we set a key with expiration = 24 hours. Next time if it fails, we check for the key existence and either skip reporting or create a new one. If the query stops failing the key will be removed in 24 hours.

Reply:
Which is what I wrote, but instead I offered a weekly/monthly expiry :)

The meaning of a daily expiry is that silenced reports (those who past the reporting threshold) will renew daily. This means that you may receive, for example, 10 e-mails a day, with the last few e-mails warning you that this will soon be silenced, and then the next morning you will start receiving them again.

Don't you think a weekly expiry is less annoying?

Reply:
I suggested the rate limit/silencing as an alternative to the counter. This way the user will receive only a single report per day for each query.